Library/Homebrew/utils.rb

```diff
@@ -506,21 +506,48 @@ def initialize(error)
     end
   end
 
-  def open(url, &_block)
-    # This is a no-op if the user is opting out of using the GitHub API.
-    return if ENV["HOMEBREW_NO_GITHUB_API"]
+  def api_token
+    @api_token ||= if ENV["HOMEBREW_GITHUB_API_TOKEN"]
+      ENV["HOMEBREW_GITHUB_API_TOKEN"]
+    else
+      github_credentials = IO.popen(["git", "credential", "fill"], "w+") do |io|
+        io.puts "protocol=https\nhost=github.com"
+        io.close_write
+        io.read
+      end
+      /username=(.+)/ =~ github_credentials
+      github_username = $1
+      /password=(.+)/ =~ github_credentials
+      github_password = $1
 
-    require "net/https"
+      [github_password, github_username] if github_username && github_password
+    end
+  end
 
+  def api_headers
     headers = {
       "User-Agent" => HOMEBREW_USER_AGENT,
       "Accept"     => "application/vnd.github.v3+json"
     }
+    token, username = api_token
+    if token && !token.empty?
+      if username && !username.empty?
```

Comment:
In that case it becomes `api_credential_headers` I think which I'd rather avoid; I can see a wider use than that.